lib/sharezone_widgets/lib/src/state_sheet/dialog.dart

```diff
@@ -28,34 +33,47 @@ class StateDialog {
     required Future<bool> future,
     Duration delay = const Duration(milliseconds: 250),
   }) async {
-    final dialogPop = show(context);
+    BuildContext? dialogContext;
+
+    final dialogPop = _show(
+      context,
+      onReceivedContext: (context) => dialogContext = context,
```

Comment:
```suggestion
      onDialogContextCreated: (context) => dialogContext = context,
```

I think all names are confusing because this is kinda hacky, but I personally think that this might a little bit less confusing.